demo/api/src/content-generation/content-generation.service.ts

```diff
@@ -0,0 +1,29 @@
+import { ContentGenerationServiceInterface, OpenAiContentGenerationConfig, OpenAiContentGenerationService } from "@comet/cms-api";
+import { Inject, Injectable } from "@nestjs/common";
+import { Config } from "@src/config/config";
+import { CONFIG } from "@src/config/config.module";
+
+const isValidOpenAiConfig = (config: Partial<OpenAiContentGenerationConfig>): config is OpenAiContentGenerationConfig => {
+    return Boolean(config.apiKey && config.apiUrl && config.deploymentId);
+};
+
+@Injectable()
+export class ContentGenerationService implements ContentGenerationServiceInterface {
+    openAiContentGenerationServiceConfig: OpenAiContentGenerationConfig;
+
+    constructor(@Inject(CONFIG) private readonly config: Config, private readonly openAiContentGenerationService: OpenAiContentGenerationService) {
+        if (isValidOpenAiConfig(config.contentGeneration)) {
+            this.openAiContentGenerationServiceConfig = config.contentGeneration;
+        } else {
+            throw new Error("Found invalid contentGeneration config");
+        }
+    }
+
+    async generateAltText(fileUrl: string) {
+        return this.openAiContentGenerationService.generateAltText(fileUrl, this.openAiContentGenerationServiceConfig);
```

Comment:
I don't know about all the details, but seeing this it feels strange that you have to pass the configuration of a service into it for every call.

Reply:
One big advantage of this approach is that the `AzureOpenAiContentGenerationService` can be extended in the future without breaking changes because each method gets passed the config it needs

Reply:
hm, you never know what will happen in future, I'd care more about the present...

additional options/arguments are always possible (for new methods)

Reply:
@nsams what would you prefer instead? A `init` method?